src/components/common/pills/pills.tsx

```diff
@@ -1,145 +1,102 @@
-                disableDelete={disableDeleteButton}
-            />
-        ));
 
-    return (
-        <DndContext
-            sensors={sensors}
-            collisionDetection={closestCenter}
-            onDragEnd={handleDragEnd}
-        >
-            <SortableContext
-                items={fields}
-                strategy={verticalListSortingStrategy}
-            >
-                <Form.Label>Pill Items</Form.Label>
-                <div>{fields.length > 0 && renderPillItems()}</div>
-                <AddMultiEntryButton
-                    icon={<PlusIcon />}
-                    styleType="light"
-                    onClick={handleAddButtonClick}
-                    role="button"
-                    type="button"
-                >
-                    Add option
-                </AddMultiEntryButton>
-            </SortableContext>
-            <Controller
-                name="pillPosition"
-                control={control}
-                render={({ field }) => (
-                    <TogglePair
-                        label={{
-                            mainLabel: "List position",
-                            subLabel:
-                                "This displays pills on top or bottom of the text area.",
-                        }}
-                        value={field.value === "top"}
-                        options={[
-                            { icon: <TopPositionIcon />, title: "Top" },
-                            { icon: <BottomPositionIcon />, title: "Bottom" },
-                        ]}
-                        onChange={(value) =>
-                            field.onChange(value ? "top" : "bottom")
+    const renderPillsToggle = () => (
+        <Controller
+            name="pills"
+            control={control}
+            defaultValue={false}
+            render={({ field }) => (
+                <TogglePair
+                    label={{
+                        mainLabel: "Pills",
+                        subLabel:
+                            "This allows a list of selectable short texts to display in pill form. This helps participant to fill up text area fast with less typing.",
+                    }}
+                    value={field.value}
+                    onChange={(value) => {
+                        if (!value) {
+                            /** Without this, the form remains dirtied when toggling Pills to true and back to false */
+                            unregister(["pillItems", "pillPosition"]);
+                        } else {
+                            setValue("pillItems", [
+                                {
+                                    content: "",
+                                },
+                                {
+                                    content: "",
+                                },
+                            ]);
+                            setValue("pillPosition", "top");
```

Comment:
after some experimenting I believe the issue is due to our use of `setValue()`
- `setValue` appears to dirty form if the value does not match the default values
- this applies even if `shouldDirty` is false
- even with unregister, the form is considered dirty; it does not "un-dirty" it
- need to change default value via `resetField`
- caveat 1: you cant do that for `useFieldArray`, official stance is it is not a single field hence it is not supported
- caveat 2: field needs to be registered first before you can reset it, probably need setTimeout

I will propose to explore using `resetField` for pillPosition whereas we will continue with this hack for `pillItems`, do leave a more detailed explanation to why we are using this workaround.